modules/react/layout/lib/utils/grid.ts

```diff
@@ -0,0 +1,93 @@
+import {Property} from 'csstype';
+
+/** style props to for grid container properties */
+export type GridStyleProps = {
+  /** sets `align-items` property */
+  alignItems?: Property.AlignItems;
+  /** sets `align-content` property */
+  alignContent?: Property.AlignContent;
+  /**
+   * sets `display` property
+   * @default 'grid'
+   * */
+  display?: 'grid' | 'inline-grid';
+  /** sets `justify-items` property */
+  justifyItems?: Property.JustifyItems;
+  /** sets `justify-content` property */
+  justifyContent?: Property.JustifyContent;
+
+  // Container Types
+  /** sets `grid-template` property */
+  gridTemplate?: Property.GridTemplate;
+  /** sets `grid-template-area` property */
+  gridTemplateAreas?: Property.GridTemplateAreas;
+  /** sets `grid-template-columns` property */
+  gridTemplateColumns?: Property.GridTemplateColumns;
+  /** sets `grid-template-rows` property */
+  gridTemplateRows?: Property.GridTemplateRows;
+  /** sets `gap` property */
+  gridGap?: Property.GridGap;
+  /** sets `column-gap` property */
+  gridColumnGap?: Property.GridColumnGap;
+  /** sets `row-gap` property */
+  gridRowGap?: Property.GridRowGap;
+  /** sets `place-items` property */
+  gridPlaceItems?: Property.PlaceItems;
+  /** sets `grid-auto-columns` property */
+  gridAutoColumns?: Property.GridAutoColumns;
+  /** sets `grid-auto-rows` property */
+  gridAutoRows?: Property.GridAutoRows;
+  /** sets `grid-area` property */
+  gridArea?: Property.GridArea;
+  /** sets `grid-auto-flow` property */
+  gridAutoFlow?: Property.GridAutoFlow;
+  /** sets `grid` property */
+  grid?: Property.Grid;
+};
+
+const gridProps = {
+  alignContent: 'alignContent',
+  alignItems: 'alignItems',
+  display: 'display',
+  justifyContent: 'justifyContent',
+  justifyItems: 'justifyItems',
+  gridTemplateAreas: 'gridTemplateAreas',
+  gridTemplate: 'gridTemplate',
+  gridTemplateColumns: 'gridTemplateColumns',
+  gridTemplateRows: 'gridTemplateRows',
+  gridGap: 'gridGap',
```

Comment:
While using these `-Gap` props, I realized it would be really nice if they knew about our space tokens so I could write:

```tsx
<Grid gridGap="s">
```

But let's follow up this PR with that functionality. 👍🏻

Reply:
Added an issue. #1703